tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1360,4 +1361,43 @@ public function commentsProvider() : array
             'valid with extra opening brackets' => ['(DC2Type:should((.stop)).before)', 'should((.stop', $currentType],
         ];
     }
+
+    public function testCreateAndListSequences() : void
+    {
+        if ( ! $this->_sm->getDatabasePlatform()->supportsSequences()) {
+            $this->markTestSkipped('This test is only supported on platforms that support sequences.');
+        }
+
+        $sequence1Name           = 'sequence_1';
+        $sequence1AllocationSize = 1;
+        $sequence1InitialValue   = 1;
+        $sequence2Name           = 'sequence_2';
+        $sequence2AllocationSize = 2;
+        $sequence2InitialValue   = 2;
+        $sequence1               = new Sequence($sequence1Name, $sequence1AllocationSize, $sequence1InitialValue);
+        $sequence2               = new Sequence($sequence2Name, $sequence2AllocationSize, $sequence2InitialValue);
+
+        $this->_sm->createSequence($sequence1);
+        $this->_sm->createSequence($sequence2);
+        /** @var Sequence[] $actualSequences */
+        $actualSequences = [];
+        foreach ($this->_sm->listSequences() as $sequence) {
+            $actualSequences[$sequence->getName()] = $sequence;
+        }
+
+        $isOracle      = $this->_sm->getDatabasePlatform()->getName() === OraclePlatform::NAME;
+        $sequence1Name = $isOracle ? strtoupper($sequence1Name) : $sequence1Name;
+        $sequence2Name = $isOracle ? strtoupper($sequence2Name) : $sequence2Name;
+
+        $actualSequence1 = $actualSequences[$sequence1Name];
+        $actualSequence2 = $actualSequences[$sequence2Name];
+
+        $this->assertSame($sequence1Name, $actualSequence1->getName());
+        $this->assertEquals($sequence1AllocationSize, $actualSequence1->getAllocationSize());
+        $this->assertEquals($sequence1InitialValue, $actualSequence1->getInitialValue());
+
+        $this->assertSame($sequence2Name, $actualSequence2->getName());
+        $this->assertEquals($sequence2AllocationSize, $actualSequence2->getAllocationSize());
+        $this->assertEquals($sequence2InitialValue, $actualSequence2->getInitialValue());
```

Comment:
Not sure the assertions are intentionally this verbose, but the same could be checked in a more compact way:
```php
$sequence1 = new Sequence('sequence_1', 1, 2);
$sequence2 = new Sequence('sequence_2', 3, 4);

// create the sequences

self::assertArraySubset([
    $sequence1,
    $sequence2,
], $this->_sm->listSequences());

Reply:
I'm fan of verbosity but can remake if you think it better suits code style of this project. Should I proceed then? :)

Reply:
Depending on what's the purpose of this verbosity. The approach I suggested is more state-based. It only tests that the collection returned by `listSequences()` contains the objects which were previously created. Isn't it what it's supposed to do?

The current approach also tests the public API of the `Sequence` class. For instance, if there was some logic in `getInitialValue` and `getAllocationSize()`, the test would have to take this logic into account as well which would make it less stable.

There are no strict coding guidelines in this regard, so it's up to you how you want to structure the test.

Reply:
I agree. However, assertArraySubset asserts by keys and there are other Sequences created in db as well.

![image](https://user-images.githubusercontent.com/327717/34343609-4e32c772-e9d6-11e7-85fe-e3477b5f3823.png)

![image](https://user-images.githubusercontent.com/327717/34343610-55c601ac-e9d6-11e7-8e69-ec65d43f5b78.png)

I reworked it a bit

Reply:
It's failing on Psql 9.5. I'd have to either sort it by name or revert to the previous verbose version.

Reply:
Eh… I didn't know `assertArraySubset()` is strict about numeric keys. The last approach could fail on any DB because the order of sequences is not enforced.

The problem is not verbosity itself, the problem is also that having the logic like `array_map()` or `array_filter()` outside of assertions which makes the test less readable and less clear about which exactly case it covers.

How about this:
```php
$list = $sm->listSequences();
self::assertContains($sequence1, $list);
self::assertContains($sequence2, $list);
```

Reply:
That wouldn't work as those are different objects.

I guess in the end this is the simplier solution:

```php
$this->assertSame($sequence1Name, $actualSequence1->getName());
$this->assertEquals($sequence1AllocationSize, $actualSequence1->getAllocationSize());
$this->assertEquals($sequence1InitialValue, $actualSequence1->getInitialValue());
```

Reply:
Okay, let's leave it like this.

Reply:
Kk, it's there